build/common.props

```diff
@@ -1,7 +1,7 @@
 <Project>
   <PropertyGroup>
     <!-- Extensions can have independent versions and only increment when released -->
-    <Version>0.0.1-alpha</Version>
+    <Version>0.0.1-alpha$(VersionSuffix)</Version>
```

Comment:
Can you enlighten me on where this variable value is set? Would love to know more.

Reply:
This is not appear on the code. It is on the pipeline. As you suggest, I create a pipeline for publishing nuget package underdevelopment.  `$(VersionSuffix)` will be ignored for the pipeline if you don't set anything. For the production nuget deployment, you don't need to care it. No effect for the main pipeline. 

However, under development, you might want to use nuget package. (I needed because I'm writing tutorial with AppService plan) . You can do it pack it, save to the LocalNuget and clear cache ... and so on. Rather than that, we can create a simple pipeline for enable it to publish it to the local users (not nuget.org) with build.id versioning as you said.  This PR enable this one. 

The pipeline configuration is like this. You can find VersionSuffix configuration and result. For more detail, you can see the AzureDevOps of ours.  

In case you create a pipeline for pushing nuget package for developers, the pipeline is like this.  This pipeline push nuget package to the Azure DevOps  artifact. 
![image](https://user-images.githubusercontent.com/1390976/55642816-67c97180-5786-11e9-8157-5589be9c3c37.png)

Build task
![image](https://user-images.githubusercontent.com/1390976/55642888-90ea0200-5786-11e9-8ad6-0293858a3815.png)

Pack task
![image](https://user-images.githubusercontent.com/1390976/55642930-ad863a00-5786-11e9-879b-460ff0e779d0.png)

Then publish. On the Artifact, you can see this result. You will see the version is updated by the build id.  You can user it from your VS. It might be handly. 
![image](https://user-images.githubusercontent.com/1390976/55643013-eaeac780-5786-11e9-9fef-e4a1e19995da.png)

Reply:
very nice! thanks for explaining!